src/datachain/cli/commands/storage/base.py

```diff
@@ -0,0 +1,80 @@
+from typing import TYPE_CHECKING
+
+from datachain.error import DataChainError
+
+if TYPE_CHECKING:
+    from argparse import Namespace
+
+    from fsspec import AbstractFileSystem
+
+    from datachain.catalog import Catalog
+    from datachain.client.fsspec import Client
+
+
+class StorageImplementation:
+    def __init__(self, args: "Namespace", catalog: "Catalog"):
```

Comment:
let's not use argc concepts down the stack. It should be used only on the level of parsing args

Reply:
This is not down the stack. This is the helper functions inside the cli commands module itself.